api/v1beta1/types.go

```diff
@@ -114,6 +114,21 @@ type NetworkSpec struct {
 	LoadBalancerBackendPort *int32 `json:"loadBalancerBackendPort,omitempty"`
 }
 
+// LoadBalancerSpec contains configuration for one or more LoadBalancers.
+type LoadBalancerSpec struct {
+
+	// APIServerInstanceGroupTagOverride overrides the default setting for the
+	// tag used when creating the API Server Instance Group.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=16
+	// +kubebuilder:validation:Pattern=`(^[1-9][0-9]{0,31}$)|(^[a-z][a-z0-9-]{4,28}[a-z0-9]$)`
+	// +optional
+	APIServerInstanceGroupTagOverride *string `json:"apiServerInstanceGroupTagOverride,omitempty"`
```

Comment:
With the way the API is structured I'm not sure it makes sense for the InstanceGroupTagOverride to belong to the load balancer spec. To me it seems like we should have an instance group spec and define a field in the cluster spec for a a ControlPlaneInstanceGroup. 

I think the issue is, for example, if we add a capability for a secondary load balancer, how do we handle this field potentially being set twice? Instance groups seem more like a dependency of the load balancer.

Reply:
To provide some context on plan for a follow-on PR to enable Internal and External LBs,  the `APIServerInstanceGroupTagOverride` would not actually be set twice. I was going to add a `LoadBalancerConfig`:
```
type LoadBalancingConfig {
 		LBScheme: External|Internal (default=External)
  		LBType: Passthrough|proxy (default=Proxy)
}
```
then LoadBalancerSpec would be
```
type LoadBalancerSpec struct {
	APIServerInstanceGroupTagOverride *string 
    InternalLBConfig *LoadBalancingConfig
    ExternalLBConfig *LoadBalancingConfig
}
```
This keeps the Override with other `APIServerInstanceGroupTagOverride` configuration and allows Internal or External LBs to be created (or maintains current configuration as the External LB will be created if neither config is set)

Reply:
I had assumed there would be a separate spec for each LB. Thanks for clarifying and this LGTM